### PR TITLE
Check for `is_numeric()` instead of `is_scalar()` on IntegerType

### PR DESCRIPTION
**Pull request by sdustinh**
Reference: https://github.com/cakephp/cakephp/issues/12509

### CODE REVIEW

File: src/Database/Type/IntegerType.php [hunk 1]
```diff
@@ -98,7 +98,7 @@ public function toPHP($value, Driver $driver)
     public function manyToPHP(array $values, array $fields, Driver $driver)
     {
         foreach ($fields as $field) {
-            if (!isset($values[$field])) {
+            if (!isset($values[$field]) || !is_numeric($values[$field])) {
```
Comment (lorenzo) [inline src/Database/Type/IntegerType.php]: It should throw instead of drop the value

Comment (markstory) [reply src/Database/Type/IntegerType.php]: :+1: we drop values in marshalling, and raise when converting to PHP or DB values.

**Comment by sdustinh** [comment]
@lorenzo @markstory If I'm going to throw the exception for non-numeric values in `manyToPHP()`, do you want me to emulate `toDatabase()` and `continue` for `null` and `''` or throw the exception for them as well?

**Comment by markstory** [comment]
`null` and `''` should be cast into null and don't require exceptions.

**Comment by markstory** [comment]
There are a few tests that are failing. Those tests will need to be updated before this can be merged in.

**Comment by sdustinh** [comment]
:+1: Just wanted to make sure we were all okay with the changes before fixing the tests.

File: src/Database/Type/IntegerType.php [hunk 2]
```diff
@@ -101,6 +101,14 @@ public function manyToPHP(array $values, array $fields, Driver $driver)
             if (!isset($values[$field])) {
                 continue;
             }
+
+            if ($values[$field] !== null && $values[$field] !== '' && !is_numeric($values[$field])) {
```
Comment (lorenzo) [inline src/Database/Type/IntegerType.php]: * no need for `$values[$field] !== null` there is an `isset` check for the value a few lines above

* no need for `$values[$field] !== ''` the `is_numeric`, we want the `is_numeric` check to catch it

Comment (sdustinh) [reply src/Database/Type/IntegerType.php]: :+1: removed them

**Pull request merged**